src/Renci.SshNet/ShellStream.cs

```diff
@@ -36,6 +36,22 @@ public class ShellStream : Stream
         /// </summary>
         public event EventHandler<ExceptionEventArgs> ErrorOccurred;
 
+        /// <summary>
+        /// Occurs when a disconnect occured.
+        /// </summary>
+        public event EventHandler<EventArgs> DisconnectOccured;
+        
+        /// <summary>
+        /// Whether the instance is disposed.
+        /// </summary>
+        public bool Disposed
```

Comment:
While I'm not a fan of exposing the "Disposed" state of an object (the User API should have complete control of that), this was added to check for a disposed state since an unexpected channel closure could cause the ShellStream to Dispose without explicit user API calls. Should I change the event handler for _channel disconect to not Dispose the ShellStream? Essentially, an abrupt channel closure puts the ShellStream in a 'half-used'/'half-disposed' state. Let me know your thoughts.